packages/core/src/ui/UIGroup.ts

```
@@ -0,0 +1,215 @@
+        const replaced = preParentGroup._disorderedGroups.deleteByIndex(this._groupIndex);
+        replaced && (replaced._groupIndex = this._groupIndex);
+        this._groupIndex = -1;
+      }
+      if (parentGroup) {
+        const disorderedGroups = parentGroup._disorderedGroups;
+        this._groupIndex = disorderedGroups.length;
+        disorderedGroups.add(this);
+      }
+      const preParentGroupEntity = this._parentGroupEntity;
+      if (preParentGroupEntity !== parentGroupEntity) {
+        preParentGroupEntity && preParentGroupEntity._unRegisterModifyListener(this._onParentEntityModify);
```

Biome

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)